woocommerce/woocommerce-gutenberg-products-block
# Product Collection: Add stock status filter

This PR adds `Stock status` control to the Product collection block. 
<img width="267" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/a84b4dcf-caf6-4292-8636-de2f1a139365">




Fixes #9362 

### Testing

1. Create a new post
2. Add Product Collection block
3. With the `Product Collection` block selected, check the sidebar for the `Stock status` control. Ensure that it is displayed as shown below:
<img width="267" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/a84b4dcf-caf6-4292-8636-de2f1a139365">

4. Test the functionality of the stock status filter by modifying its selection. Observe that the changes should immediately reflect in the Editor view. For instance, if `On backorder` is the selected option, only products with the `On backorder` status should be visible in the Editor.
5. Save and publish the post.
    - Navigate to the live post on the Frontend and verify that the changes you made in the Editor are accurately reflected.
6. Go back to the Editor and confirm that the filters can be reset using the `Reset` or `Reset all` button, as shown below:
<img width="279" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/dbea8bfc-2d6c-4ece-9236-4b8bf176f5e3">


### WooCommerce Visibility




### Changelog

> Product Collection: Add stock status control to sidebar settings
